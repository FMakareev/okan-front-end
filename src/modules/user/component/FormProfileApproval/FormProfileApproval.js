import React, { Component, Fragment } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

/**Components */
import Text from '../../../../components/Text/Text';
import Flex from '../../../../components/Flex/Flex';
import Tbody from '../../../../components/Table/Tbody';
import Td from '../../../../components/Table/Td';
import Tr from '../../../../components/Table/Tr';

const TrStyled = styled(Tr)`
  display: flex;
  justify-content: space-around;
  padding: 3px 0;
  cursor: pointer;

  &:hover {
    background-color: #007faf21;
  }
`;

const TbodyStyled = styled(Tbody)`
  border: 1px solid #848484;
  border-radius: 5px;
  margin: 10px 0 0 0;
`;

export class FormProfileApproval extends Component {
  static propTypes = {
    /** name document */
    name: PropTypes.string,
    /** number document*/
    number: PropTypes.string,
  };

  constructor(props) {
    super(props);
    this.state = {};
  }
  render() {
    const {
      initialValues
    } = this.props;

    return (
      <Fragment>
        <Text
          fontSize={6}
          lineHeight={8}
          color={'color7'}
          textAlign={'center'}
          mb={[13]}
          fontFamily={'primary500'}>
          На согласование
        </Text>
        <Tbody>
          <Flex justifyContent={'space-around'}>
            <Text fontSize={6} lineHeight={8} fontFamily={'primary500'}>
              Номер документа
            </Text>

            <Text fontSize={6} lineHeight={8} fontFamily={'primary500'}>
              Название документа
            </Text>
          </Flex>
        </Tbody>
        {/*initialValues.map(item=>item.name)*/}
        <TbodyStyled>
          <TrStyled>
            <Td fontFamily={'primary500'}>{'1'}</Td>
            <Td fontFamily={'secondaryBold'}>{'2'}</Td>
          </TrStyled>
        </TbodyStyled>

        <TbodyStyled>
          <TrStyled>
            <Td fontFamily={'primary500'}>{'1'}</Td>
            <Td fontFamily={'secondaryBold'}>{'2'}</Td>
          </TrStyled>
        </TbodyStyled>
      </Fragment>
    );
  }
}

export default FormProfileApproval;
