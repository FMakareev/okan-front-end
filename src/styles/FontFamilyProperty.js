/**
 * @params {string} fontFamily - string name of the font name property in the theme
 * @params {object} theme
 * */
export const FontFamilyProperty = ({ fontFamily, theme }) => {
  try {
    if (fontFamily && theme.fontFamily) {
      return `font-family: '${theme.fontFamily[fontFamily]}';`;
    }
  } catch (e) {
    console.error(e);
  }
  return undefined;
};
export default FontFamilyProperty;
