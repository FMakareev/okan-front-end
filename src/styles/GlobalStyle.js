import { createGlobalStyle } from 'styled-components'


export const GlobalStyle = createGlobalStyle`
  *{
    box-sizing: border-box;
  }
  body, html, #app {
    padding: 0;
    margin: 0;
    width: 100%;
    min-height: 100vh;
    background-color: #fff;
  }
`;
