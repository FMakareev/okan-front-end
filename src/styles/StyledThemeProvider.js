import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'rebass';

/** Variants */
import { ButtonVariant } from './variants/buttonVariant';
import { ButtonSize } from './variants/buttonSize';
import { ButtonComment } from './variants/buttonComment';
import { TextVariant } from './variants/textVariant';
import { InputSize } from '@lib/styles/variants/InputSize';
import { InputVariant } from '@lib/styles/variants/InputVariant';

export const ColorPallet = {
  color0: '#FFFFFF',
  color1: '#E5E5E5',
  color2: '#DEDEDE',
  color3: '#4F4F4F',
  color4: '#848484',
  color5: '#007FAF',
  color6: '#006699',
  color7: '#00649C',
  color8: '#007CC4',
  color9: '#F97E63',
  color10: '#AEAEAE',
  color11: '#333333',
  color12: '#DF4624',
  color13: '#f2f2f2',
  color14: '#fff3c1',
};

const Space = [
  0,
  2,
  4,
  8,
  12,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  68,
  72,
  76,
  80,
];

const Weight = [300, 500, 700];

const Breakpoints = ['1024px', '1200px', '1440px'];

const boxShadow = [
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 2px 4px 0px rgba(0,127,175,1)',
];

const textAlign = ['left', 'center', 'right'];

export const StyledThemeProvider = ({ children }) => (
  <Provider
    theme={{
      space: Space,
      fontSizes: Space,
      fontWeight: Weight,
      lineHeight: Space,
      boxShadow,
      borderRadius: Space,
      borderColor: ColorPallet,
      breakpoints: Breakpoints,
      colors: ColorPallet,
      variant: {
        buttons: ButtonVariant,
        buttonSize: ButtonSize,
        buttonComment: ButtonComment,
        text: TextVariant,
        inputSize: InputSize,
        inputVariant: InputVariant,
      },
      fontFamily: {
        primary300: 'Museo Sans 300',
        primary500: 'Museo Sans 500',
        primary700: 'Museo Sans 700',
        secondary: 'Circe Regular',
        secondaryBold: 'Circe Bold',
      },
      textAlign: textAlign,
    }}>
    {children}
  </Provider>
);

StyledThemeProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

StyledThemeProvider.defaultProps = {
  children: 'children is empty',
};

export default StyledThemeProvider;
