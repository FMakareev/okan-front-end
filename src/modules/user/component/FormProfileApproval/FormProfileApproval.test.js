import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { StyledThemeProvider } from '../../../../styles/StyledThemeProvider';
import { FormProfileApproval } from './FormProfileApproval';

describe('News container initial', () => {
  const props = {
    data: {
      name: '1',
      number: '2',
    },
  };

  const newsContainer = shallow(
    <StyledThemeProvider>
      <FormProfileApproval {...props} />
    </StyledThemeProvider>,
  ); // передали props

  it('renders properly', () => {
    expect(newsContainer).toMatchSnapshot();
  });
});
