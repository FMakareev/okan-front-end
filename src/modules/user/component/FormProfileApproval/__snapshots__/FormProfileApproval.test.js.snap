// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News container initial renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyledThemeProvider>
    <FormProfileApproval
      data={
        Object {
          "name": "1",
          "number": "2",
        }
      }
    />
  </StyledThemeProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <FormProfileApproval
        data={
          Object {
            "name": "1",
            "number": "2",
          }
        }
      />,
      "theme": Object {
        "borderColor": Object {
          "color0": "#FFFFFF",
          "color1": "#E5E5E5",
          "color10": "#AEAEAE",
          "color11": "#333333",
          "color12": "#DF4624",
          "color13": "#f2f2f2",
          "color2": "#DEDEDE",
          "color3": "#4F4F4F",
          "color4": "#848484",
          "color5": "#007FAF",
          "color6": "#006699",
          "color7": "#00649C",
          "color8": "#007CC4",
          "color9": "#F97E63",
        },
        "borderRadius": Array [
          0,
          2,
          4,
          8,
          12,
          16,
          18,
          20,
          24,
          28,
          32,
          36,
          40,
          44,
          48,
          52,
          56,
          60,
          64,
          68,
          72,
          76,
          80,
        ],
        "boxShadow": Array [
          "2px 4px 8px 0px rgba(138,138,138,0.5)",
          "2px 4px 8px 0px rgba(138,138,138,0.5)",
          "2px 2px 4px 0px rgba(0,127,175,1)",
        ],
        "breakpoints": Array [
          "1024px",
          "1200px",
          "1440px",
        ],
        "colors": Object {
          "color0": "#FFFFFF",
          "color1": "#E5E5E5",
          "color10": "#AEAEAE",
          "color11": "#333333",
          "color12": "#DF4624",
          "color13": "#f2f2f2",
          "color2": "#DEDEDE",
          "color3": "#4F4F4F",
          "color4": "#848484",
          "color5": "#007FAF",
          "color6": "#006699",
          "color7": "#00649C",
          "color8": "#007CC4",
          "color9": "#F97E63",
        },
        "fontFamily": Object {
          "primary300": "Museo Sans 300",
          "primary500": "Museo Sans 500",
          "primary700": "Museo Sans 700",
          "secondary": "Circe Regular",
          "secondaryBold": "Circe Bold",
        },
        "fontSizes": Array [
          0,
          2,
          4,
          8,
          12,
          16,
          18,
          20,
          24,
          28,
          32,
          36,
          40,
          44,
          48,
          52,
          56,
          60,
          64,
          68,
          72,
          76,
          80,
        ],
        "fontWeight": Array [
          300,
          500,
          700,
        ],
        "lineHeight": Array [
          0,
          2,
          4,
          8,
          12,
          16,
          18,
          20,
          24,
          28,
          32,
          36,
          40,
          44,
          48,
          52,
          56,
          60,
          64,
          68,
          72,
          76,
          80,
        ],
        "space": Array [
          0,
          2,
          4,
          8,
          12,
          16,
          18,
          20,
          24,
          28,
          32,
          36,
          40,
          44,
          48,
          52,
          56,
          60,
          64,
          68,
          72,
          76,
          80,
        ],
        "textAlign": Array [
          "left",
          "center",
          "right",
        ],
        "variant": Object {
          "buttonSize": Object {
            "large": Object {
              "fontFamily": "Circe Bold",
              "fontSize": " 32px",
              "lineHeight": "40px",
              "minWidth": "500px",
              "padding": "8px 60px",
            },
            "medium": Object {
              "fontFamily": "Circe Bold",
              "fontSize": "18px",
              "lineHeight": "24px",
              "minWidth": "250px",
              "padding": "8px 0",
            },
            "small": Object {
              "fontFamily": "Circe Regular",
              "fontSize": " 16px",
              "lineHeight": "16px",
              "minWidth": "200px",
              "padding": "2px 0",
            },
            "xsmall": Object {
              "fontSize": " 14px",
              "fontWeight": 700,
              "lineHeight": "16px",
              "minWidth": "70px",
              "padding": "2px 0",
            },
          },
          "buttons": Object {
            "empty": Object {
              "backgroundColor": "#ffffff",
              "border": 0,
            },
            "large": Object {
              "&:hover": Object {
                "backgroundColor": "#00649C",
                "color": "#ffffff",
                "fill": "#ffffff",
              },
              "backgroundColor": "#ffffff",
              "border": "2px solid",
              "color": "#00649C",
              "fill": "#00649C",
            },
            "small": Object {
              "&:hover": Object {
                "backgroundColor": "#848484",
                "color": "#ffffff",
                "fill": "#ffffff",
              },
              "backgroundColor": "#fff",
              "border": "1px solid",
              "color": "#848484",
              "fill": "#848484",
            },
            "xsmall": Object {
              "&:hover": Object {
                "backgroundColor": "#fff",
                "color": "#00649C",
                "fill": "#00649C",
              },
              "backgroundColor": "#00649C",
              "border": "1px solid #00649C",
              "color": "#fff",
              "fill": "#fff",
            },
          },
          "text": Object {
            "document": Object {
              "color": "#4F4F4F",
              "fontFamily": "'Museo Sans 300'",
              "fontSize": "14px",
              "lineHeight": "20px",
              "marginBottom": "10px",
              "marginTop": "10px",
              "textAlign": "left",
            },
            "firstHeader": Object {
              "color": "#333333",
              "fontFamily": "'Museo Sans 700'",
              "fontSize": "18px",
              "lineHeight": "30px",
              "marginBottom": "10px",
              "marginTop": "10px",
              "textAlign": "left",
            },
            "secondHeader": Object {
              "color": "#333333",
              "fontFamily": "'Museo Sans 500'",
              "fontSize": "14px",
              "lineHeight": "20px",
              "marginBottom": "10px",
              "marginTop": "10px",
              "textAlign": "left",
            },
          },
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "data": Object {
          "name": "1",
          "number": "2",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <FormProfileApproval
          data={
            Object {
              "name": "1",
              "number": "2",
            }
          }
        />,
        "theme": Object {
          "borderColor": Object {
            "color0": "#FFFFFF",
            "color1": "#E5E5E5",
            "color10": "#AEAEAE",
            "color11": "#333333",
            "color12": "#DF4624",
            "color13": "#f2f2f2",
            "color2": "#DEDEDE",
            "color3": "#4F4F4F",
            "color4": "#848484",
            "color5": "#007FAF",
            "color6": "#006699",
            "color7": "#00649C",
            "color8": "#007CC4",
            "color9": "#F97E63",
          },
          "borderRadius": Array [
            0,
            2,
            4,
            8,
            12,
            16,
            18,
            20,
            24,
            28,
            32,
            36,
            40,
            44,
            48,
            52,
            56,
            60,
            64,
            68,
            72,
            76,
            80,
          ],
          "boxShadow": Array [
            "2px 4px 8px 0px rgba(138,138,138,0.5)",
            "2px 4px 8px 0px rgba(138,138,138,0.5)",
            "2px 2px 4px 0px rgba(0,127,175,1)",
          ],
          "breakpoints": Array [
            "1024px",
            "1200px",
            "1440px",
          ],
          "colors": Object {
            "color0": "#FFFFFF",
            "color1": "#E5E5E5",
            "color10": "#AEAEAE",
            "color11": "#333333",
            "color12": "#DF4624",
            "color13": "#f2f2f2",
            "color2": "#DEDEDE",
            "color3": "#4F4F4F",
            "color4": "#848484",
            "color5": "#007FAF",
            "color6": "#006699",
            "color7": "#00649C",
            "color8": "#007CC4",
            "color9": "#F97E63",
          },
          "fontFamily": Object {
            "primary300": "Museo Sans 300",
            "primary500": "Museo Sans 500",
            "primary700": "Museo Sans 700",
            "secondary": "Circe Regular",
            "secondaryBold": "Circe Bold",
          },
          "fontSizes": Array [
            0,
            2,
            4,
            8,
            12,
            16,
            18,
            20,
            24,
            28,
            32,
            36,
            40,
            44,
            48,
            52,
            56,
            60,
            64,
            68,
            72,
            76,
            80,
          ],
          "fontWeight": Array [
            300,
            500,
            700,
          ],
          "lineHeight": Array [
            0,
            2,
            4,
            8,
            12,
            16,
            18,
            20,
            24,
            28,
            32,
            36,
            40,
            44,
            48,
            52,
            56,
            60,
            64,
            68,
            72,
            76,
            80,
          ],
          "space": Array [
            0,
            2,
            4,
            8,
            12,
            16,
            18,
            20,
            24,
            28,
            32,
            36,
            40,
            44,
            48,
            52,
            56,
            60,
            64,
            68,
            72,
            76,
            80,
          ],
          "textAlign": Array [
            "left",
            "center",
            "right",
          ],
          "variant": Object {
            "buttonSize": Object {
              "large": Object {
                "fontFamily": "Circe Bold",
                "fontSize": " 32px",
                "lineHeight": "40px",
                "minWidth": "500px",
                "padding": "8px 60px",
              },
              "medium": Object {
                "fontFamily": "Circe Bold",
                "fontSize": "18px",
                "lineHeight": "24px",
                "minWidth": "250px",
                "padding": "8px 0",
              },
              "small": Object {
                "fontFamily": "Circe Regular",
                "fontSize": " 16px",
                "lineHeight": "16px",
                "minWidth": "200px",
                "padding": "2px 0",
              },
              "xsmall": Object {
                "fontSize": " 14px",
                "fontWeight": 700,
                "lineHeight": "16px",
                "minWidth": "70px",
                "padding": "2px 0",
              },
            },
            "buttons": Object {
              "empty": Object {
                "backgroundColor": "#ffffff",
                "border": 0,
              },
              "large": Object {
                "&:hover": Object {
                  "backgroundColor": "#00649C",
                  "color": "#ffffff",
                  "fill": "#ffffff",
                },
                "backgroundColor": "#ffffff",
                "border": "2px solid",
                "color": "#00649C",
                "fill": "#00649C",
              },
              "small": Object {
                "&:hover": Object {
                  "backgroundColor": "#848484",
                  "color": "#ffffff",
                  "fill": "#ffffff",
                },
                "backgroundColor": "#fff",
                "border": "1px solid",
                "color": "#848484",
                "fill": "#848484",
              },
              "xsmall": Object {
                "&:hover": Object {
                  "backgroundColor": "#fff",
                  "color": "#00649C",
                  "fill": "#00649C",
                },
                "backgroundColor": "#00649C",
                "border": "1px solid #00649C",
                "color": "#fff",
                "fill": "#fff",
              },
            },
            "text": Object {
              "document": Object {
                "color": "#4F4F4F",
                "fontFamily": "'Museo Sans 300'",
                "fontSize": "14px",
                "lineHeight": "20px",
                "marginBottom": "10px",
                "marginTop": "10px",
                "textAlign": "left",
              },
              "firstHeader": Object {
                "color": "#333333",
                "fontFamily": "'Museo Sans 700'",
                "fontSize": "18px",
                "lineHeight": "30px",
                "marginBottom": "10px",
                "marginTop": "10px",
                "textAlign": "left",
              },
              "secondHeader": Object {
                "color": "#333333",
                "fontFamily": "'Museo Sans 500'",
                "fontSize": "14px",
                "lineHeight": "20px",
                "marginBottom": "10px",
                "marginTop": "10px",
                "textAlign": "left",
              },
            },
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Object {
            "name": "1",
            "number": "2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
